### PR TITLE
test: create e2e tests for basic spending plan limit

### LINKED ISSUE #3065: [HBAR Rate Limit Redesign] Create e2e tests for basic spending plan limit
```[tasklist]
### Tasks
- [x] Create a test ensuring a user not included in a plan is treated as a BASIC user
- [x] Create a test simulating transactions that accumulate to just below the limit
- [x] Create a test simulating transactions that exceed the limit
- [x] Create a test deploying X amount of contracts for a BASIC user until limited
```

### PR DESCRIPTION
**Pull request by konstantinabl**
**Description**:

This PR adds acceptance tests to verify the scenarios listed in the issue that is linked

**Related issue(s)**:

Fixes #3065 

**Notes for reviewer**:

### CODE REVIEW

File: packages/server/tests/acceptance/hbarLimiter.spec.ts [hunk 1]
```diff
@@ -392,49 +388,41 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
...
           expect(ethAddressSpendingPlanRepository.findByAddress(accounts[2].address, requestDetails)).to.be.rejected;
-          expect(remainingHbarsBefore).to.be.gt(0);
-          let spendingPlanId;
           try {
             for (let i = 0; i < 50; i++) {
               await deployContract(largeContractJson, accounts[2].wallet);
-              const remainingHbars = Number(await metrics.get(testConstants.METRICS.REMAINING_HBAR_LIMIT));
-
-              // Check that the BASIC spending plan is created after the first deployment only
-              if (i === 0) {
```
Comment (konstantinabl) [inline packages/server/tests/acceptance/hbarLimiter.spec.ts]: @quiet-node Why is this deleted? I wanted to have this check here to ensure we create a basic plan as intended?

File: packages/server/tests/acceptance/hbarLimiter.spec.ts [hunk 2]
```diff
@@ -392,49 +388,41 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
...
-                const ethSpendingPlan = await ethAddressSpendingPlanRepository.findByAddress(
-                  accounts[2].wallet.address,
-                  requestDetails,
-                );
-
-                spendingPlanId = ethSpendingPlan.planId;
-              }
-              const hbarPlan = await hbarSpendingPlanRepository.findByIdWithDetails(spendingPlanId, requestDetails);
-
-              expect(hbarPlan.amountSpent).to.be.gt(spentToday);
```
Comment (konstantinabl) [inline packages/server/tests/acceptance/hbarLimiter.spec.ts]: @quiet-node And why remove this? I think it is still separate than what you are doing in the catch? 
I want to ensure that in each call we calculate the amount spent correctly and we increase it?

File: packages/server/tests/acceptance/hbarLimiter.spec.ts [hunk 3]
```diff
@@ -309,7 +338,24 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
...
 
-      describe('Rate Limit', () => {
+      describe('Tiered HBAR Rate Limit', () => {
+        before(async () => {
+          // Clear the cache from all BASIC plans
+          const basicPlans = await hbarSpendingPlanRepository.findAllActiveBySubscriptionTier(
+            [SubscriptionTier.BASIC],
+            requestDetails,
+          );
+          for (const plan of basicPlans) {
```
Comment (konstantinabl) [inline packages/server/tests/acceptance/hbarLimiter.spec.ts]: Why delete all the plans :?

File: packages/relay/src/utils.ts [hunk 4]
```diff
@@ -67,4 +67,42 @@ export class Utils {
...
+   * @returns {number} The estimated transaction fee in tinybars.
+   */
+  public static estimateFileTransactionsFee(
+    callDataSize: number,
+    fileChunkSize: number,
+    currentNetworkExchangeRateInCents: number,
+  ): number {
+    const fileCreateTransactions = 1;
+    const fileCreateFeeInCents = constants.NETWORK_FEES_IN_CENTS.FILE_CREATE_PER_5_KB;
+
```
Comment (ebadiere) [inline packages/relay/src/utils.ts:87]: Do we need to validate that `fileChunkSize` is not a zero value to prevent divide by zero issues in the next line?
```
  if (fileChunkSize <= 0) {
    throw new Error('fileChunkSize must be greater than zero.');
  }
  ```

File: packages/relay/src/utils.ts [hunk 5]
```diff
@@ -67,4 +67,42 @@ export class Utils {
...
+    // The first chunk goes in with FileCreateTransaciton, the rest are FileAppendTransactions
+    const fileAppendTransactions = Math.floor(callDataSize / fileChunkSize) - 1;
```
Comment (ebadiere) [inline packages/relay/src/utils.ts:89]: Perhaps I am missing something here, but I think we want to use `Math.ceil` here instead of `.floor` Using `.floor` will round down to the nearest integer, possibly missing the last chunk.  `.ceil` will round up to the nearest integer catching the last chunk even if the `callDataSize` is not an exact multiple of the `fileChunkSize`.

File: packages/server/tests/acceptance/hbarLimiter.spec.ts [hunk 6]
```diff
@@ -370,66 +415,40 @@ describe('@hbarlimiter HBAR Limiter Acceptance Tests', function () {
...
 
-          await new Promise((r) => setTimeout(r, 3000));
           const spendingPlanAssociatedAfterSecond = await hbarSpendingPlanRepository.findByIdWithDetails(
             ethSpendingPlan.planId,
             requestDetails,
           );
           expect(amountSpendAfterFirst).to.be.lt(spendingPlanAssociatedAfterSecond.amountSpent);
-        });
 
-        it('should eventually exhaust the hbar limit for a BASIC user after multiple deployments of large contracts', async function () {
```
[the remark this answers is not in the document]
Comment (victor-yanev) [reply packages/server/tests/acceptance/hbarLimiter.spec.ts]: Discussed offline

File: packages/server/tests/acceptance/hbarLimiter.spec.ts (hunk 3 above)
Comment (victor-yanev) [reply packages/server/tests/acceptance/hbarLimiter.spec.ts]: Discussed offline.

File: packages/relay/src/utils.ts (hunk 5 above)
Comment (victor-yanev) [reply packages/relay/src/utils.ts:89]: I think this logic is outdated, @quiet-node can you verify

**Comment by quiet-node** [comment]
closed in light of https://github.com/hashgraph/hedera-json-rpc-relay/pull/3104
